File: src/main/kotlin/com/github/immersingeducation/immersingpicker/core/History.kt
package com.github.immersingeducation.immersingpicker.core

import java.time.LocalDateTime

/**
 * 历史记录类，用于表示班级的历史记录
 * @param createTime 历史记录创建时间
 * @param selector 这次记录是用哪个抽选器创建的
 * @param students 历史记录中的学生列表
 * @author CC想当百大
 * @since v1.0.0.a
 */
data class History(
    val createTime: LocalDateTime = LocalDateTime.now(),
    val selector: String,
    val students: List<Student>
)

File: ImmersingPicker.Core/Models/History.cs
namespace ImmersingPicker.Core.Models;

public class History
{
    public DateTime CreateTime { get; set; }
    public string Selector { get; set; }
    public List<Student> Students { get; set; }

    public History()
    {
        CreateTime = DateTime.Now;
        Selector = string.Empty;
        Students = new List<Student>();
    }

    public History(DateTime createTime, string selector, List<Student> students)
    {
        CreateTime = createTime;
        Selector = selector;
        Students = students;
    }
}